src/compiler/compilation.jl

```diff
@@ -18,15 +19,19 @@ const CUDACompilerJob = CompilerJob{PTXCompilerTarget,CUDACompilerParams}
 GPUCompiler.runtime_module(@nospecialize(job::CUDACompilerJob)) = CUDA
 
 # filter out functions from libdevice and cudadevrt
-GPUCompiler.isintrinsic(@nospecialize(job::CUDACompilerJob), fn::String) =
-    invoke(GPUCompiler.isintrinsic,
-           Tuple{CompilerJob{PTXCompilerTarget}, typeof(fn)},
-           job, fn) ||
-    fn == "__nvvm_reflect" || startswith(fn, "cuda")
+function GPUCompiler.isintrinsic(@nospecialize(job::CUDACompilerJob), fn::String)
+    is_intrinsic = invoke(GPUCompiler.isintrinsic,
+                    Tuple{CompilerJob{PTXCompilerTarget}, typeof(fn)}, job, fn)
+    is_intrinsic |= fn == "__nvvm_reflect"
+    is_intrinsic |= startswith(fn, "cuda")
+    is_intrinsic |= !job.config.params.link_libdevice ? startswith(fn, "__nv_") : false # Reactant.jl wants to handle __nv_ functions
```

Comment:
This is pretty unreadable.

I also don't understand why this should be handled here. Doesn't Reactant have its own compiler job type to implement `isintrinsic` for? Or make sure the intrinsics are lowered _before_ calling into GPUCompiler.jl validation?

Reply:
No Reactant does not have a compiler job (it overlays a few functions of cuda.jl and imports from cuda.jl directly)